tests/models/test_simple_nn.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Sequence
+
+import jax
+import jax.numpy as jnp
+import pytest
+from infra.module_tester import ComparisonMetric, test, test_with_random_inputs
+from infra.test_model import TestModel
+
+
+class SimpleNN(TestModel):  # TODO what's the benefit of inheriting nnx.Module?
+    # TODO upgrade to python 3.13 to enable this decorator
+    # @override
+    def __call__(
+        self, act: jax.Array, w0: jax.Array, b0: jax.Array, w1: jax.Array, b1: jax.Array
+    ) -> jax.Array:
+        x = jnp.matmul(act, w0) + b0
+        x = jnp.matmul(x, w1) + b1
+        return x
+
+    # @override
+    @staticmethod
+    def get_model() -> TestModel:
+        return SimpleNN()
+
+    # @override
+    @staticmethod
+    def get_model_inputs() -> Sequence[jax.Array]:
+        act_shape, w0_shape, b0_shape, w1_shape, b1_shape = (
+            (32, 784),
+            (784, 128),
+            (1, 128),
+            (128, 128),
+            (1, 128),
+        )
+
+        act = jax.numpy.ones(act_shape)
+        w0 = jax.numpy.ones(w0_shape)
+        b0 = jax.numpy.ones(b0_shape)
+        w1 = jax.numpy.ones(w1_shape)
+        b1 = jax.numpy.zeros(b1_shape)
+
+        return [act, w0, b0, w1, b1]
+
+
+def test_simple_nn():
+    model = SimpleNN.get_model()
+    inputs = SimpleNN.get_model_inputs()
+
+    assert test(model, inputs, ComparisonMetric.PCC)
```

Comment:
This should all happen inside the ModelTester. Our task when writing tests should only be to implement the `load_model` and `load_inputs` and then when we run the tester it should automatically load the model, load the inputs, compile it, run it in eval/train modes, verify results.